pkg/solana/client/client.go

```diff
@@ -242,27 +242,22 @@ func (c *Client) ChainID(ctx context.Context) (mn.StringID, error) {
 
 	ctx, cancel := context.WithTimeout(ctx, c.contextDuration)
 	defer cancel()
+
 	v, err, _ := c.requestGroup.Do("GetGenesisHash", func() (interface{}, error) {
 		return c.rpc.GetGenesisHash(ctx)
 	})
 	if err != nil {
 		return "", err
 	}
-	hash := v.(solana.Hash)
-
-	var network string
-	switch hash.String() {
-	case DevnetGenesisHash:
-		network = "devnet"
-	case TestnetGenesisHash:
-		network = "testnet"
-	case MainnetGenesisHash:
-		network = "mainnet"
-	default:
-		c.log.Warnf("unknown genesis hash - assuming solana chain is 'localnet'")
-		network = "localnet"
+
+	hash := v.(solana.Hash).String()
+
+	// If the genesis hash does not match any known network, we return it as a 'localnet' instead of a genesis hash
```

Comment:
nit: Also comment that genesis hash of localnet changes on each run, thus it cannot be used as a validation, and instead needs a static name like "localnet".